### DISCUSSION TITLE
Stray urgency hint on workspace

### DISCUSSION THREAD

**Discussion by joepvd** [i3/i3#4861] [Q&A]
I am using a wrapper around rofi to do the following:

I launch an application. If the application is already open, I want it to move to the current workspace. If it is not yet open, I want it to get open. The script relies on `focus_on_window_activation` to be set to `smart`. I use this a lot with my email client, Thunderbird.

<details><summary>This is the code:</summary>

```sh
#!/usr/bin/env bash

set -euo pipefail

listen_for_changes() {
  con_id=$(
    while IFS= read -r line; do
      jq --exit-status --raw-output '
        if .change == "urgent" then .container.id
        elif .change == "focus" then "0"
        else halt_error end
      ' 2>/dev/null <<<"$line" &&
        break
    done < <(
      i3-msg -t subscribe -m '["window"]'
    )
  )
  (( con_id == 0 )) && return
  i3-msg "[con_id=$con_id] move workspace current, focus" >/dev/null 2>&1
}; export -f listen_for_changes

bash -c listen_for_changes &

rofi -i -show drun -modi drun -show-icons -theme /usr/share/rofi/themes/solarized_alternate.rasi
```
</details>

The problem:

This leaves a stray urgency hint on the workspace where the application used to be. 

Questions:
- Is this maybe a bug in i3? I'd expect that urgency hints on workspaces get cleared when none of the containers inside the workspace have an urgency hint set.
- Am I doing something wrong? Should I be waiting for ticks or so?
- Is there a way to clear an urgency hint from a container (which happens to be a workspace) with `i3-msg`?

**Comment by Airblader** [comment]
I just tried to reproduce this in another way by opening a terminal and running

```
# Focus on another workspace during the first sleep
# echo -e "\a" is a bell, in my terminal this sets the urgency hint
i3 mark "foo"; sleep 3; echo -e "\a"; sleep 3; i3 "[con_mark=foo] move window to workspace otherws"
```

This seemed to work correctly: the workspace with the window became urgent, and another three seconds later it was moved to "otherws" along with the urgency hint.

> Is this maybe a bug in i3?

I think so, yes. I would suggest submitting a bug report, ideally with a vanilla config and of course a log file. Thanks!

> Is there a way to clear an urgency hint from a container (which happens to be a workspace) with i3-msg?

There's no i3-level APIs, no.
